### PR TITLE
Config Defaults

### PR DESCRIPTION
**Pull request by synbot**
### Acceptance Criteria
1. Default config values are stored in `/application/config.js`
1. Only changed values are loaded from the environment configs
1. Uses `lodash.merge` for deep merging

### CODE REVIEW

**Comment by areida** [comment]
:star:

File: application/config/qa.js [hunk 1]
```diff
@@ -8,8 +8,6 @@ backend = __BACKEND__ || '%QA_API_HOST%';
 
 module.exports = {
     api : {
-        hostname  : backend,
-        port      : 80,
-        client_id : '123'
+        url : backend
```
Comment (paulstatezny) [inline application/config/qa.js]: This should be `hostname` in each file.

**Comment by paulstatezny** [comment]
:interrobang:

**Comment by areida** [comment]
:star:

File: application/config.js [hunk 2]
```diff
@@ -2,21 +2,42 @@
...
+        hostname  : 'localhost',
+        port      : 9000,
+        oauth     : {
+            logout : '/api/oauth/logout',
+            token  : '/api/oauth/token'
+        },
+        prefix : '/api',
+        proxy  : {
+            hostname : ''
+        }
```
Comment (paulstatezny) [inline application/config.js]: This includes changes from synapsestudios/synapse-common#40 and #98.

**Comment by paulstatezny** [comment]
:interrobang:

**Comment by paulstatezny** [comment]
:+1:

**Pull request merged**